qutebrowser/browser/shared.py

```diff
@@ -310,7 +310,12 @@ def netrc_authentication(url, authenticator):
     user, password = None, None
     try:
         net = netrc.netrc(config.val.content.netrc_file)
-        authenticators = net.authenticators(url.host())
+        if url.port():
```

Comment:
Sorry, didn't notice this before: The default/invalid value for `QUrl.port()` is -1, because 0 is a valid (but special) port. Doing `if url.port():` lets `-1` ("no port") through, so this should probably be `if url.port() != -1:` instead.